File: uploadservice/src/main/java/net/gotev/uploadservice/schemehandlers/SchemeHandler.java
package net.gotev.uploadservice.schemehandlers;

import android.content.Context;

import java.io.FileNotFoundException;
import java.io.InputStream;

/**
 * Allows for different file representations to be used by abstracting several characteristics
 * and operations
 * @author stephentuso
 * @author gotev
 */
public interface SchemeHandler {
    void init(String path);
    long getLength(Context context);
    InputStream getInputStream(Context context) throws FileNotFoundException;
    String getContentType(Context context);
    String getName(Context context);
}

File: uploadservice/src/main/java/net/gotev/uploadservice/schemehandlers/SchemeHandler.kt
package net.gotev.uploadservice.schemehandlers

import android.content.Context
import java.io.InputStream

interface SchemeHandler {
    /**
     * Initialize instance with file path.
     */
    fun init(path: String)

    /**
     * Gets file size in bytes.
     */
    fun size(context: Context): Long

    /**
     * Gets file input stream to read it.
     */
    fun stream(context: Context): InputStream

    /**
     * Gets file content type.
     */
    fun contentType(context: Context): String

    /**
     * Gets file name.
     */
    fun name(context: Context): String

    /**
     * Deletes the file.
     */
    fun delete(context: Context): Boolean
}
